aten/src/ATen/native/mps/operations/BinaryOps.mm

```diff
@@ -35,7 +35,26 @@ void binaryOpTensor(const Tensor& self, const Tensor& other, const Scalar& alpha
 
   const bool is_self_scalar = self.dim() == 0;
   const bool is_other_scalar = other.dim() == 0;
-
+  bool needsScatter = false;
+
+  Tensor output_;
+  // Determine if this is an in-place operation
+  if (self.storage().data() == output.storage().data() || other.storage().data() == output.storage().data()) {
+    IValue selfIVal(self);
+    IValue otherIVal(other);
+    IValue outputIVal(output);
+
+    if (selfIVal.isAliasOf(outputIVal) || otherIVal.isAliasOf(outputIVal)) {
```

Comment:
Why not just make the output contiguous instead, and assign results back to original output at the end?

Reply:
I think if we do contiguous we'll still have to do scatter to it eventually - I think empty_mps might be a little bit less costly, but would be good to get some numbers from here. I needed scatter for this usecase: ` a_mps[:, 0] += b_mps[:, 0]`

Reply:
Yes, I agree, we shouldn't making them contiguous and incur the copy overhead. We will have to scatter back which will be slower but not as much as copy. Imagine bigger tensors there will cause lot of slow-down.